src/main/java/tool/Storage.java

```diff
@@ -0,0 +1,161 @@
+        } catch(IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    /**
+     * Loads tasks from the .txt file
+     * @param commands
+     * @return the updated array list of tasks
+     */
+    public ArrayList<Task> load(ArrayList<Task> commands) {
+        try {
+            FileReader fr = new FileReader(this.filePath);
+            BufferedReader br = new BufferedReader(fr);
+            String x;
+            int counter = 1;
+            while((x = br.readLine()) != null) {
+                if (counter == 1) {
+                    System.out.println("Here are the tasks in your list:");
+                }
+                String[] data = x.split("/");
+                if (data[0].equals("T")) {
+                    Todo t = new Todo(data[2]);
+                    if (data[1].equals("1")) {
+                        t.markAsDone();
+                    }
+                    commands.add(t);
+                    System.out.println(counter + ". " + t);
+                    counter++;
+                } else if (data[0].equals("D")) {
+                    Deadline d = new Deadline(data[2], new DateTime(data[3]));
+                    if (data[1].equals("1")) {
+                        d.markAsDone();
+                    }
+                    commands.add(d);
+                    System.out.println(counter + ". " + d);
+                    counter++;
+                } else if (data[0].equals("E")){
+                    Event e = new Event(data[2], new DateTime(data[3]));
+                    if (data[1].equals("1")) {
+                        e.markAsDone();
+                    }
+                    commands.add(e);
+                    System.out.println(counter + ". " + e);
+                    counter++;
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return commands;
+    }
+
+    /**
+     * Saves the latest task in the .txt file
+     * @param Task t
+     */
+    public void save(Task t) {
+        try {
+            this.fw.append(t.storageString() + System.lineSeparator());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void delete(int i) {
+        try {
+            FileReader fr = new FileReader(filePath);
+            BufferedReader br = new BufferedReader(fr);
+            FileWriter ff = new FileWriter("src/main/java/data/temp.txt", true);
+            int j = 1;
+            String x;
+            while ((x = br.readLine()) != null) {
+                if (i != j) {
+                    ff.write(x + System.lineSeparator());
+                    ff.flush();
+                }
+                j++;
+            }
+            ff.close();
+            Files.copy(Paths.get("src/main/java/data/temp.txt"), Paths.get(filePath), StandardCopyOption.REPLACE_EXISTING);
+            Files.delete(Paths.get("src/main/java/data/temp.txt"));
```

Comment:
Since all the storage-modifying logic does the same process of writing and re-saving the file, I think it would be better to pull the common logic into one single method.